Add a Helmchart for GeoHealthCheck

stranljip opened the pull request:
This PR adds a Helm chart for GeoHealth check which allows an easy deployment on Kubernetes.

stranljip committed:
code formatting & linter

Signed-off-by: heidmann <carsten.heidmann@disy.net>

stranljip committed:
initial version Helmchart

Signed-off-by: heidmann <carsten.heidmann@disy.net>

stranljip committed:
first attempt to add custom certificates

Signed-off-by: heidmann <carsten.heidmann@disy.net>

stranljip committed:
current state of things

Signed-off-by: heidmann <carsten.heidmann@disy.net>

stranljip committed:
working chart

Signed-off-by: heidmann <carsten.heidmann@disy.net>

stranljip committed:
add documentation to the values

Signed-off-by: heidmann <carsten.heidmann@disy.net>

stranljip committed:
add persistenmce for SQLite database

Signed-off-by: heidmann <carsten.heidmann@disy.net>

stranljip committed:
change directory name

stranljip committed:
update docs

stranljip commented:
Closes #483

stranljip committed:
add possibility to add a update strategy

stranljip committed:
change to boolean values

stranljip committed:
remove file from first draft

stranljip committed:
fix strategy type and set it to recreate

stranljip committed:
fix the definition for the update strategy

stranljip commented:
Hi - are there any contributors in this project who can have a look at my PR or is this project unmaintained and I should continue the work solely on my fork?

justb4 commented:
Sorry @stranljip . This project is still maintained, for example I run several production sites, including geoqos.com, only the maintainers have little time. Also for most the project has enough maturity to guard day-to-day geospatial webservices.

I have some experience in K8s, but not with Helm Charts. But I see all changes are under the `charts` directory, not affecting existing functionality. And very useful. So we could also just broadly review and merge your PR and trust you have the proper knowledge to maintain in the future. 
My only comment would be: naming the dir `charts` could be confusing, as within GHC we also draw status charts. Is it an idea to name the dir `k8s`? Or does Helm apply and the files under `charts` apply to more than K8s?
Ok @tomkralidis ?

tomkralidis commented:
> Sorry @stranljip . This project is still maintained, for example I run several production sites, including geoqos.com, only the maintainers have little time. Also for most the project has enough maturity to guard day-to-day geospatial webservices.
> 
> I have some experience in K8s, but not with Helm Charts. But I see all changes are under the `charts` directory, not affecting existing functionality. And very useful. So we could also just broadly review and merge your PR and trust you have the proper knowledge to maintain in the future. My only comment would be: naming the dir `charts` could be confusing, as within GHC we also draw status charts. Is it an idea to name the dir `k8s`? Or does Helm apply and the files under `charts` apply to more than K8s? Ok @tomkralidis ?

I'd probably create `docker/helm` and `docker/kubernetes` accordingly.

tomkralidis commented on charts/geohealthcheck/.editorconfig:
Remove file

stranljip replied:
Removed from PR

tomkralidis commented on charts/geohealthcheck/.yamllint:
Is this file required?

stranljip replied:
Not necessarily. If you don't have linting configurations elsewhere in your repo I can remove it

stranljip replied:
Removed from PR

tomkralidis commented on charts/geohealthcheck/templates/NOTES.txt:
Rename to `README.txt` or is this file required?

stranljip replied:
This is required, the file `NOTES.txt` is used as a template for the console output after creating a Helm release.

tomkralidis reviewed (changes requested):
Some minor questions, thanks for this contribution!

I would also suggest adding a new page in the docs on this functionality, as well as a `Makefile` as a convenience.

stranljip commented:
> > Sorry @stranljip . This project is still maintained, for example I run several production sites, including geoqos.com, only the maintainers have little time. Also for most the project has enough maturity to guard day-to-day geospatial webservices.
> > I have some experience in K8s, but not with Helm Charts. But I see all changes are under the `charts` directory, not affecting existing functionality. And very useful. So we could also just broadly review and merge your PR and trust you have the proper knowledge to maintain in the future. My only comment would be: naming the dir `charts` could be confusing, as within GHC we also draw status charts. Is it an idea to name the dir `k8s`? Or does Helm apply and the files under `charts` apply to more than K8s? Ok @tomkralidis ?
> 
> I'd probably create `docker/helm` and `docker/kubernetes` accordingly.

The Helm chart does only apply to k8s. `docker/helm` and `docker/kubernetes` do not "sound" quite right but I don't want to mess up your directory structure but I would suggest to refactor it a little bit:
* `docker` --> `install/docker-compose`
* `charts` --> `install/kubernetes/charts` or `charts/geohealthcheck` --> `install/kubernetes/helmchart`

stranljip committed:
review: remove unwanted files

stranljip committed:
doc: add minimal documentation

tomkralidis commented:
> > I would also suggest adding a new page in the docs on this functionality, as well as a `Makefile` as a convenience.
> 
> I have to admit that I am completely unfamiliar with this framework. Is your suggestion to add a Makefile in the Helmchart directory and add one page about the Helmchart or should I add a page in your `docs` dir?

Yes, add a `Makefile` so anyone wanting to run Helm has some help.  For docs, thinking more, move the content of `charts/docs/installhelm.rst` (of this PR) into `docs/install.rst` add a **section** to page `docs/install.rst`.

stranljip commented:
Sorry for the long silence here. I will start working on this issue in the next weeks and add the requested changes.

stranljip commented:
I have now added a minimal section on installation with Helm on Kubernetes in `install.rst` and I have added a more comprehensive README.md in a `docs` subdirectory in the charts directory, following your example for Docker. I have not created a Makefile - as far as I understand it this is only for building the documentation and is obsolete for me since I integrated it with your docs.
